src/v/datalake/schema_registry.cc

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "pandaproxy/schema_registry/types.h"
+#include "datalake/schema_registry.h"
+
+namespace datalake {
+get_schema_id_result get_value_schema_id(const iobuf& buf) {
+    if (buf.size_bytes() < sizeof(uint8_t) + sizeof(int32_t)) {
+        return get_schema_error::not_enough_bytes;
+    }
+    iobuf_const_parser parser(buf);
+    auto magic = parser.consume_type<uint8_t>();
+    if (magic != 0) {
```

Comment:
is this a value we control?

Reply:
This is the schema registry format leading "magic byte".

Some named constants would help in reading the code here